### ISSUE TITLE
PannerNode always crashes

### ISSUE THREAD

**Issue by ggadwa** [orottier/web-audio-api-rs#369]
Machine: AMD, 16 GB, Windows 10 Home, Rust 1.7, web-audio 0.33.0

I knew web audio from a javascript 3d development environment I made, so thank you for this project!  While working in rust, calling this anywhere in my code:

let panner = ctx.create_panner();

Will eventually cause (usually when it's called a second time):

thread 'cpal_wasapi_out' panicked at 'called `Option::unwrap()` on a `None` value', C:\Users\ggadwa\.cargo\registry\src\index.crates.io-6f17d22bba15001f\web-audio-api-0.33.0\src\render\graph.rs:261:14

Note the panner is not connected to an output or connected from an input source, just calling create_panner() causes this.  Note that if I instead use Panner::new with options the same thing happens, but more immediately (just when I call it the first time, or at least in my testing.)

Further notes: Creating a source, connecting it to an output, and playing it within the same code work perfectly; but within the same code creating a panner node will crash it, so it seems related specific to the panner node.

As I am new at Rust, it is possible this is some mistake on my part, i.e., a lifetime/de-allocation problem.

**Comment by orottier** [comment]
Hi @ggadwa, thanks for flagging this issue with us.
It definitely sounds like an issue in our library.
However I'm not directly able to reproduce it, could you share the full code snippet you are running?

**Comment by orottier** [comment]
Thanks again.
I have managed to shrink the snippet to this, and now I can reproduce the error:

```rust
    let context = AudioContext::default();
    let panner = context.create_panner();
    drop(panner);
    println!("sleep");
    std::thread::sleep(std::time::Duration::from_secs(1));
    let mut panner = context.create_panner();
    println!("sleep");
    std::thread::sleep(std::time::Duration::from_secs(1));
```

For some reason, the `drop` and `sleeps` are necessary for the panic condition to occur.
I will have a look at it now, the `PannerNode` has some special treatment going on (the `AudioListener` is not added to the audio graph when no panners are present, for performance) and I suspect it has something to do with it.

**Comment by ggadwa** [comment]
Sorry for the big snippet, I wasn't sure what was potentially the problem.  Note that in my game code, I am creating (and updating) a listener, and that code doesn't crash anything, without the panner create, if that helps at all.

Thanks for this library, and glad I could help with a bug report!

**Comment by orottier** [comment]
No worries, your snippet helped me narrow down the search.

I have actually found the issue: whenever a new AudioPanner is added, we connect the AudioListener to it in the audio graph to make sure the panner renderer has access to the listener position/orientation. But we don't account for the panner to go out of scope and drop from the audio graph. The listener will still have a connection to it which can no longer be resolved. This makes the application crash. I'm working on a fix now

**Comment by orottier** [comment]
I have prepared a fix in #370 
@b-ma is there anything you would like to add to the discussion?
If not, I will probably merge this tomorrow and put out a new release

**Comment by b-ma** [comment]
Hey, no anything particular to add on my side, except thanks @ggadwa for the feedback

> If not, I will probably merge this tomorrow and put out a new release

ok (I might need some little help to figure out how to fix https://github.com/ircam-ismm/node-web-audio-api/pull/37 then)

**Comment by orottier** [comment]
I have released version 0.34 just now.

@ggadwa there are some breaking changes in this release. Mainly most audio nodes must be mutable now, this is simply done changing
```rust
- let src = ctx.create_buffer_source();
+ let mut src = ctx.create_buffer_source();
```

@b-ma sorry I missed your call for help over at the node bindings. I will have a look. It will be a bit tricky indeed
